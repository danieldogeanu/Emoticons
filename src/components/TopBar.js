import React from 'react';
import './TopBar.scss';

function TopBar(props) {
	return (
		<div className="TopBar">
			<div className="wrapper">
				{props.children}
			</div>
		</div>
	);
}

export default TopBar;