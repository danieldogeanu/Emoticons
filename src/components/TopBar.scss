@import '../styles/includes';

.TopBar {
	display: flex;
	justify-content: center;
	align-items: center;
	position: fixed;
	top: 0; left: 0; right: 0;
	z-index: 200;
	width: 100%;
	height: rem(70);
	padding: 0 rem(10);
	background-color: color-scheme(background);
	@include box-shadow(0, 0, rem(100), 0, color(black), 0.15);
	@include breakpoint(min, rem(301)) { height: rem(70); padding: 0 rem(20); }
	@include breakpoint(min, rem(481)) { height: rem(90); padding: 0 rem(40); }
	@include breakpoint(min, rem(601)) { height: rem(100); padding: 0 rem(60); }

	.wrapper {
		display: block;
		width: 100%;
		max-width: rem(730);
	}
}