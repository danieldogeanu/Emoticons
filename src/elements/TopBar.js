import React from 'react';
import '../styles/components/TopBar.scss';

const TopBar = (props) => (
	<div className="TopBar" data-testid="TopBar">
		<div className="wrapper">
			{props.children}
		</div>
	</div>
);

export default TopBar;
