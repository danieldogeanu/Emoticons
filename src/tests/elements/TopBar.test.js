import React from 'react';
import {render} from '@testing-library/react';
import TopBar from '../../elements/TopBar';

describe('TopBar Element', () => {

	it('renders top bar properly', () => {
		const testId = 'TopBar';
		const childText = 'TopBar Test Child';
		const {getByTestId, getByText} = render(<TopBar>{childText}</TopBar>);
		const renderedTopBar = getByTestId(testId);
		const renderedWrapper = getByText(childText);

		expect(renderedTopBar).toBeInTheDocument();
		expect(renderedTopBar).toHaveClass(testId);
		expect(renderedTopBar).toContainElement(renderedWrapper);
		expect(renderedTopBar).toMatchSnapshot();
		expect(renderedWrapper).toHaveTextContent(childText);
		expect(renderedWrapper).toHaveClass('wrapper');
	});

});
