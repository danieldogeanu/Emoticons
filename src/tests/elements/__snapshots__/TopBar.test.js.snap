// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBar Element renders top bar properly 1`] = `
<div
  class="TopBar"
  data-testid="TopBar"
>
  <div
    class="wrapper"
  >
    TopBar Test Child
  </div>
</div>
`;
